lib/improver/utilities/cube_manipulation.py

```diff
@@ -856,6 +921,30 @@ def enforce_float32_precision(input_cubes):
                 cube.data = cube.data.astype(np.float32)
 
 
+def demote_float64_precision(input_cubes):
+    """Take input cube of any precision and convert any float64 data to
+    float32.
+
+    Args:
+        input_cubes (cubelist or cube):
+            List containing one or more iris cubes to test and adjust if
+            necessary.
+            Note: The code will modify the cubes in-place.
+
+    """
+    # If single cube - place within cubelist.
+    if isinstance(input_cubes, iris.cube.Cube):
+        input_cubes = iris.cube.CubeList([input_cubes])
+
+    # Cycle through the cubes
+    for cube in input_cubes:
+        assert isinstance(cube, iris.cube.Cube), 'Object is not a cube'
+
+        # Modify data if it is float64
+        if cube.data.dtype.num == np.dtype(np.float64).num:
+            cube.data = cube.data.astype(np.float32)
```

Comment:
I struggled with the dtype comparison. I can't remember if I tried your suggestion.

NO!! Don't change the coordinates - this will change date-time values!

Reply:
I reckon the cube.dtype should work safely. There are interesting problems using coord.dtype since that can represent the points dtype rather than the bounds dtype.

I think no float64 except for `time` is safe... anyway I'm altering all the tests to have that!